flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SymbolArgumentTypeStrategy.java

```diff
@@ -55,13 +55,15 @@ public SymbolArgumentTypeStrategy(Class<? extends Enum<? extends TableSymbol>> s
             }
         }
 
-        final SymbolType<?> symbolType = (SymbolType<?>) argumentType.getLogicalType();
-        if (symbolType.getSymbolClass() != symbolClass) {
+        if (!callContext.getArgumentValue(argumentPos, symbolClass).isPresent()) {
             if (throwOnFailure) {
                 throw callContext.newValidationError(
-                        "Unsupported argument symbol type. Expected symbol '%s' but actual symbol was '%s'.",
+                        "Unsupported argument symbol type. Expected symbol '%s' but actual symbol was %s.",
                         symbolClass.getSimpleName(),
-                        symbolType.getDefaultConversion().getSimpleName());
+                        callContext
+                                .getArgumentValue(argumentPos, Enum.class)
```

Comment:
I'm a bit confused, but if L58, we don't have a `symbolClass`, can we still have an Enum here?
is there a test covering this, and the "invalid" case?

Reply:
`SymbolArgumentTypeStrategyTest` is testing this path. We are in control of creating symbols so we can safely assume an `Enum` here. Everything else is highly unlikely. For the `invalid` case I would need to create an invalid expression. The alternative would have been an `InvalidArgumentException`.

Reply:
If it's easy to have a test like that, would be nice for completeness.

Reply:
No, that is not easy. As I said, I would need to create a completely invalid expression. Maybe some constructor checks would already prevent that.